src/rgw/rgw_user.cc

```diff
@@ -1391,13 +1391,18 @@ int RGWSubUserPool::add(RGWUserAdminOpState& op_state, std::string *err_msg, boo
 {
   std::string subprocess_msg;
   int ret;
+  int32_t key_type = op_state.get_key_type();
 
   ret = check_op(op_state, &subprocess_msg);
   if (ret < 0) {
     set_err_msg(err_msg, "unable to parse request, " + subprocess_msg);
     return ret;
   }
 
+  if (key_type == KEY_TYPE_S3 && op_state.get_access_key().empty()) {
```

Comment:
I think we should deal the access key and secret in the same way .Rather the secret can be created by default if we not specify the secret or the gen-secret ,but the access-key  can not .